i18n/states.php

```diff
@@ -347,8 +434,47 @@
 		'ZA' => __( 'Zamora', 'woocommerce' ),
 		'Z'  => __( 'Zaragoza', 'woocommerce' ),
 	),
-	'FI' => array(),
-	'FR' => array(),
+	'FI' => array( // Finnish regions.
+		'FI02' => __( 'Etelä-Karjalan', 'woocommerce' ),
+		'FI03' => __( 'Etelä-Pohjanmaan', 'woocommerce' ),
+		'FI04' => __( 'Etelä-Savon', 'woocommerce' ),
+		'FI05' => __( 'Kainuun', 'woocommerce' ),
+		'FI06' => __( 'Kanta-Hämeen', 'woocommerce' ),
+		'FI07' => __( 'Keski-Pohjanmaan', 'woocommerce' ),
+		'FI08' => __( 'Keski-Suomen', 'woocommerce' ),
+		'FI09' => __( 'Kymenlaakson', 'woocommerce' ),
+		'FI10' => __( 'Lapin', 'woocommerce' ),
+		'FI11' => __( 'Pirkanmaan', 'woocommerce' ),
+		'FI12' => __( 'Pohjanmaan', 'woocommerce' ),
+		'FI13' => __( 'Pohjois-Karjalan', 'woocommerce' ),
+		'FI14' => __( 'Pohjois-Pohjanmaan', 'woocommerce' ),
+		'FI15' => __( 'Pohjois-Savon', 'woocommerce' ),
+		'FI16' => __( 'Päijät-Hämeen', 'woocommerce' ),
+		'FI17' => __( 'Satakunnan', 'woocommerce' ),
+		'FI18' => __( 'Uudenmaan', 'woocommerce' ),
+		'FI19' => __( 'Varsinais-Suomen', 'woocommerce' ),
+	),
+	'FR' => array( // French regions.
+		'ARA' => __( 'Auvergne-Rhône-Alpes', 'woocommerce' ),
+		'BFC' => __( 'Bourgogne-Franche-Comté', 'woocommerce' ),
+		'BRE' => __( 'Bretagne', 'woocommerce' ),
+		'COR' => __( 'Corse', 'woocommerce' ),
+		'CVL' => __( 'Centre-Val de Loire', 'woocommerce' ),
+		'GES' => __( 'Grand Est', 'woocommerce' ),
+		'HDF' => __( 'Hauts-de-France', 'woocommerce' ),
+		'IDF' => __( 'Île-de-France', 'woocommerce' ),
+		'NAQ' => __( 'Nouvelle-Aquitaine', 'woocommerce' ),
+		'NOR' => __( 'Normandie', 'woocommerce' ),
+		'OCC' => __( 'Occitanie', 'woocommerce' ),
+		'PAC' => __( "Provence-Alpes-Côte d'Azur", 'woocommerce' ),
+		'PDL' => __( 'Pays de la Loire', 'woocommerce' ),
+	),
+	'GB' => array( // British countries/provinces.
+		'ENG' => __( 'England', 'woocommerce' ),
+		'NIR' => __( 'Northern Ireland', 'woocommerce' ),
+		'SCT' => __( 'Scotland', 'woocommerce' ),
+		'WLS' => __( 'Wales', 'woocommerce' ),
+	),
```

Comment:
One question about the county and these UK addresses:
```
2 Chestnut Hall Avenue, Maghaberry, Moira, County Armagh, BT67 0GG
40 Whitehall Way, Rockingham, Rotherham, South yorkshire, S61 4HW
19 Carisbrooke Close, EASTBOURNE, East Sussex, BN23 8EQ
```
I don't see "England", "Northern Ireland", "Scotland" or "Wales" in these addresses, do we assume the user to know which to pick? 

For example, using address `2 Chestnut Hall Avenue, Maghaberry, Moira, County Armagh, BT67 0GG` in this form, 
![image](https://user-images.githubusercontent.com/572862/73316244-94f08000-41ef-11ea-97be-6e23f9c21450.png)
I am assuming we drop the `County Armagh` entry and use `Moira` as city. Or should `County Armagh` be the county?

Reply:
Also, user made a comment on the values in `county`. https://github.com/Automattic/woocommerce-services/issues/1600#issuecomment-579532076.

Since we don't have "England", "Northern Ireland", "Scotland" or "Wales" as separate countries, any of them would be selected as "United Kingdom".  I do agree that putting these 4 countries as "county" is a bit awkward.

Reply:
There are multiple levels of subdivisions in some countries and UK is one of them. We can instead include the longer & more detailed list if that feels more natural to the UK folks, please see https://github.com/woocommerce/woocommerce/pull/25509/commits/e981529d5c16483d3910f700e16e2a369b6a6ef9

Reply:
It's very uncommon (at least in my experience spending way too much online) to select counties in the UK like this. Usually it's just a text field. In practice, it doesn't affect postal addresses much at all, since county can be derived from the postcode (postcode being the most important field for postage). To me using a select here feels unnatural.

Surprised to hear this is a problem related to taxes; there are no per-county tax rates. Everyone in the UK pays the same rate of sales tax.

Reply:
Thanks for chiming in, @mikejolley!

Background: WC Services automated taxes solution looks up tax rates based on country + state + city + zip code and since the state gets sanitized each time, it fails to find a match in WC, thus creating a new tax rule each time. In addition, the tax fails to be applied if there's a space in the County name.
This could be fixed in the WooCommerce services, however, I see they have their own custom db code to handle this case, which feels not right, so I thought we should provide a method in core to do the matching and they should then start using it. 

While it's true that in the UK everyone pays the same sales tax (although someone [mentioned it might not be true all the time](https://github.com/woocommerce/woocommerce/issues/24354#issuecomment-577882153)), I found out that even the same country+state+city+zip code combination can have multiple tax rates in some coutries, so I wasn't convinced that leaving out State when creating tax rules would be the best solution. Maybe it could work for the UK, but not in general...

One possibility I can see is to use some conditionals in WCS where for the UK the tax rules would only fill in country + zip code, while for other countries it could fill in other fields. 

TL;DR: Due to the nature of the bug, I'm not sure if there's an easier and better solution that would work in all cases, but I'm open to suggestions/finding a better solution.

Reply:
Yeah @peterfabian that is tricky. My concern is how this affects customers (checkout) as the experience will be strange and could affect conversions (we'd have no way to track that though?).

Ideally the tax rate service would return wildcards, effectively indicating that city/state is not important for the tax rate and negating the matching issue. But I don't know enough about the service so..

Reply:
As the field is optional by default, I would hope it's not going to affect the conversions.

Though another worry of mine is the huge blob of text with the list of Counties sent over at the point of checkout... it's 60kb now :/